python/pyspark/sql/column.py

```diff
@@ -712,11 +712,11 @@ def __getitem__(self, k: Any) -> "Column":
         --------
         >>> df = spark.createDataFrame([('abcedfg', {"key": "value"})], ["l", "d"])
         >>> df.select(df.l[slice(1, 3)], df.d['key']).show()
-        +------------------+------+
-        |substring(l, 1, 3)|d[key]|
-        +------------------+------+
-        |               abc| value|
-        +------------------+------+
+        +---------------+------+
+        |substr(l, 1, 3)|d[key]|
```

Comment:
To avoid this change, I think we can call `self.substring` in this method. `df.l[slice(1, 3)]` does not require a certain function name, so keeping it as it was is better.

Reply:
Ping @peter-toth

Reply:
I tried `self.substring` (see https://github.com/apache/spark/pull/42864/commits/d611fd5a8da8f0f24d54c3933b3cf7e4dbdabe2c) but there is no `self.substring` so if column name doesn't matter then let's use `self.substr`.